packages/flame/lib/src/experimental/geometry/shapes/circle.dart

```diff
@@ -82,6 +82,20 @@ class Circle extends Shape {
       ..add(_center);
   }
 
+  static final Vector2 _tmpResult = Vector2.zero();
+
+  @override
+  Vector2 nearestPoint(Vector2 point) {
+    if (_radius == 0) {
+      return _center;
+    }
+    return _tmpResult
```

Comment:
I wonder if we don't want to at least return a new vector? Could be quite strange behavior if the user saves this away and then runs this function again for another circle and then both the vectors are the result of the last one, or well they are the same object. I know we have this pattern at some places, but not sure if it is good... At the very least we should warn the user in the dartdoc that they should clone the vector if they intend to save it.

Reply:
Yeah, I have a note about it in the doc-comment.

We kinda don't have any consistency in how we approach this. There are methods that return new vectors, there are those that return existing vectors, and then those that sometimes do one thing and other times another.

It might be useful to do some benchmarking and figure out what's the actual cost of creating new vectors, and how would that compare to a "hand-crafted" Vector2 class which simply contain two doubles `x`, `y` as fields (thus, a bit like mutable `Offset`).

For now, though, we can go with this API, since it's easier to loosen up in the future if need be. Going in the other direction instead (i.e. we returned new objects, but then switched to reused objects) would be a breaking change.